project/ParadoxSupport.scala

```diff
@@ -29,23 +30,31 @@ object ParadoxSupport {
 
   class UnidocDirective(allClasses: IndexedSeq[String]) extends InlineDirective("unidoc") {
     def render(node: DirectiveNode, visitor: Visitor, printer: Printer): Unit = {
-      if (node.label.split('[')(0).contains('.')) {
-        val fqcn = node.label
+      val (directive, label, source) = node.source match {
+        case direct: Source.Direct => (s"@unidoc[${node.label}](${direct.value})", node.label, direct.value)
+        case ref: Source.Ref       => (s"@unidoc[${node.label}](${ref.value})", node.label, ref.value)
+        case Source.Empty          => (s"@unidoc[${node.label}]", node.label.split('.').last, node.label)
+      }
```

Comment:
This is to handle two different syntax:

- `@unidoc[label](source)`
- `@unidoc[lable]`

The former case is useful (e.g.) when you link to `ShardRegion$` but don't want the `$` appear in the label.
The latter shorthand syntax is convenient for the majority of the cases.